lib/semian/net_http.rb

```diff
@@ -36,6 +36,7 @@ def semian_identifier
       ::Timeout::Error, # includes ::Net::ReadTimeout and ::Net::OpenTimeout
       ::SocketError,
       ::Net::HTTPBadResponse,
+      ::Net::HTTPFatalError,
```

Comment:
Your impression in correct, it's the code within the Net HTTP request method that raises the exception.